netty/src/main/java/io/grpc/netty/NettyServer.java

```diff
@@ -140,6 +133,15 @@ public void operationComplete(ChannelFuture future) {
             ch.close();
             return;
           }
+          // `channel` shutdown can race with `ch` initialization, so this is only safe to increment
+          // inside the lock.
+          eventLoopReferenceCounter.retain();
+          ch.closeFuture().addListener(new ChannelFutureListener() {
```

Comment:
This addListener _could_ be placed after the synchronized block, but I honestly don't care.